wayland-backend/src/rs/server/common_poll.rs

```diff
@@ -88,24 +96,56 @@ impl<D> Backend<D> {
         }
     }
 
+    /// Flushes pending events destined for a client.
+    ///
+    /// If no client is specified, all pending events are flushed to all clients.
     pub fn flush(&mut self, client: Option<ClientId>) -> std::io::Result<()> {
         self.handle.flush(client)
     }
 
+    /// Returns a handle which contains the server side state of the backend.
+    ///
+    /// The handle provides a variety of functionality, such as querying information about wayland objects,
+    /// obtaining data associated with a client and it's objects, and creating globals.
     pub fn handle(&mut self) -> &mut Handle<D> {
         &mut self.handle
     }
 
+    /// Returns the underlying file descriptor.
+    ///
+    /// The file descriptor may be monitored for activity with a polling mechanism such as epoll or kqueue.
+    /// When it becomes readable, this means there are pending messages that would be dispatched if you call
+    /// [`Backend::dispatch_all_clients`].
+    ///
+    /// The file descriptor should not be used for any other purpose than monitoring it.
     pub fn poll_fd(&self) -> RawFd {
         self.poll_fd
     }
 
+    /// Dispatches all pending messages from the specified client.
+    ///
+    /// This method will not block if there are no pending messages.
+    ///
+    /// The provided `data` will be provided to the handler of messages received from the client.
+    ///
+    /// For performance reasons, use of this function should be integrated with an event loop, monitoring the
+    /// file descriptor retrieved by [`Backend::poll_fd`] and only calling this method when messages are
+    /// available.
```

Comment:
For client-specific dispatch, it's the client-specific FD that should be monitored (the one associated to its `UnixStream`).

Also, please add in the documentation that this method may dispatch other clients as well as the one that was requested when using the sys backend.